app/src/main/java/fr/free/nrw/commons/contributions/ContributionController.java

```diff
@@ -46,11 +55,118 @@ public void initiateCameraPick(Activity activity) {
+                        initiateCameraUpload(activity);
+                    }
+                },
                 R.string.storage_permission_title,
                 R.string.write_storage_permission_rationale);
     }
 
+    /**
+     * Suggest user to attach location information with pictures.
+     * If the user selects "Yes", then:
+     *
+     * Location is taken from the EXIF if the default camera application
+     * does not redact location tags.
+     *
+     * Otherwise, if the EXIF metadata does not have location information,
+     * then location captured by the app is used
+     *
+     * @param activity
+     */
+    private void askUserToAllowLocationAccess(Activity activity) {
+        DialogUtil.showAlertDialog(activity,
+            activity.getString(R.string.in_app_camera_location_permission_title),
+            activity.getString(R.string.in_app_camera_location_access_explanation),
+            activity.getString(R.string.option_allow),
+            activity.getString(R.string.option_dismiss),
+            ()-> {
+                defaultKvStore.putBoolean("inAppCameraLocationPref", true);
+                requestForLocationAccess(activity);
+            },
+            () -> {
+                defaultKvStore.putBoolean("inAppCameraLocationPref", false);
+                initiateCameraUpload(activity);
+            },
+            null,
+            true);
+    }
+
+    /**
+     * Ask for location permission if the user agrees on attaching location with pictures
+     * and the app does not have the access to location
+     *
+     * @param activity
+     */
+
+    private void requestForLocationAccess(Activity activity) {
+        PermissionUtils.checkPermissionsAndPerformAction(activity,
+            permission.ACCESS_FINE_LOCATION,
+            () -> onLocationPermissionGranted(activity),
+            () -> {},
+            R.string.ask_to_turn_location_on,
+            R.string.in_app_camera_location_permission_rationale);
+    }
+
+    /**
+     * Check if apps have access to location even after having individual access
+     *
+     * @return
+     */
+    private boolean isLocationAccessToAppsTurnedOn() {
+        return (locationManager.isNetworkProviderEnabled() || locationManager.isGPSProviderEnabled());
+    }
+
+    /**
+     * Initiate in-app camera if apps have access to location.
+     * Otherwise, show location-off dialog.
+     *
+     * @param activity
+     */
+    private void onLocationPermissionGranted(Activity activity) {
+        if (!isLocationAccessToAppsTurnedOn()) {
+            showLocationOffDialog(activity);
+        } else {
+            initiateCameraUpload(activity);
+        }
+    }
+
+    /**
+     * Ask user to grant location access to apps
+     *
+     * @param activity
+     */
+
+    private void showLocationOffDialog(Activity activity) {
+        DialogUtil
+            .showAlertDialog(activity,
+                activity.getString(R.string.location_permission_title),
```

Comment:
This should be `R.string.ask_to_turn_location_on` since this is a dialog for requesting to turn on location.